### PR TITLE
Added fp.exists() false negative shield.

### PR DESCRIPTION
**Pull request by agamemnus**
The Media API doesn't account for cases where fp.exists() may provide a false negative. (For example, APK expansions...)

Cosmetic: I also did a short-circuit on the first condition, converted to "} else {"s instead of hanging brackets, and fixed the incorrect spacing in the last few lines. (caused probably by the hanging brackets!)

### CODE REVIEW

**Comment by agamemnus** [comment]
I got the email back saying it was received.

https://issues.apache.org/jira/secure/Dashboard.jspa doesn't want to load.

**Comment by vispo** [comment]
I am still in shock reading that you can't use Android expansion files with the Cordova Media plugin to play sounds. I have a 150mb project that uses a lot of sound and I've developed it using the Media plugin. Is it true that you can't use the Media plugin with Android expansion files??? What to do???

**Comment by pniraula** [comment]
@vispo and there seems to be some problem with android 5.0 and media plugin. I am working in a project which uses media plugin to play audio. The application works well in all versions of android but 5.0.

**Comment by agamemnus** [comment]
vispo, you can use Crosswalk and do ``xwalk --disable-gesture-requirement-for-media-playback`` in ``platforms\android\assets\xwalk-command-line``.

**Comment by vispo** [comment]
Have never heard of Crosswalk before. I see it at https://crosswalk-project.org . Is it a Phonegap plugin or something entirely separate from Phonegap?

**Comment by vispo** [comment]
I hear that Amazon allows Android apps up to 150mb in size. I could probably get away without having to use an expansion file if we distribute it through Amazon rather than the Google Play Store. Have you distributed an Android app through Amazon? Is it OK? Limitations?

**Comment by agamemnus** [comment]
I have not tried using Amazon to publish my game.

Crosswalk is a system that adds a standalone version of Chromium to your Android Cordova app or game. It's only a few steps to use it via the "migrate" option and it is well-supported by Intel and a growing base of users.

**Comment by cordova-qa** [comment]
Cordova CI Build has one or more failures. 

**Commit**     - [Link](https://github.com/apache/cordova-plugin-media/pull/27/commits/6fd5193225c291a11b65e6a261b0974a30e30e8d)
**Dashboard** - [Link](http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15/)

| Builder Name  | Console Output | Test Report | Device Logs  |
|     :---:     |     :---:      |   :---:     |     :---:    |
| [Windows 8.1 Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=windows-slave,platformName=windows-8.1-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=windows-slave,platformName=windows-8.1-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=windows-slave,platformName=windows-8.1-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=windows-slave,platformName=windows-8.1-store/artifact/) |
| [Windows 10  Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=windows-slave,platformName=windows-10-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=windows-slave,platformName=windows-10-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=windows-slave,platformName=windows-10-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=windows-slave,platformName=windows-10-store/artifact/) |
| [Windows 8.1 Phone]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=windows-slave,platformName=windows-8.1-phone/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=windows-slave,platformName=windows-8.1-phone/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=windows-slave,platformName=windows-8.1-phone/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=windows-slave,platformName=windows-8.1-phone/artifact/) |
| [iOS]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=mac-slave,platformName=ios/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=mac-slave,platformName=ios/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=mac-slave,platformName=ios/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=mac-slave,platformName=ios/artifact/) |
| [Android Mac]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=mac-slave,platformName=android/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=mac-slave,platformName=android/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=mac-slave,platformName=android/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-media-pr/15//label=mac-slave,platformName=android/artifact/) |

**Comment by agamemnus** [comment]
Hard to see.... too many links....

**Comment by janpio** [comment]
It's been a loooong time @agamemnus , but can you maybe still explain what 
> cases where fp.exists() may provide a false negative. (For example, APK expansions...)
actually meant? Can you provide some context on how to test this? Thanks.

**Comment by agamemnus** [comment]
You'd have to install both my plugin and that one and you will see that the test identifies the file as not existing when it actually does, IIRC. Problem is, Google has been very bad in supporting APK expansions and they keep making changes that makes it very difficult to maintain.

**Comment by janpio** [comment]
Ok, so this is kind of an edge case? 
Anyway, I think your code itself is pretty reasonable. Can you describe exactly what is happening with the files in plain English though? From top to bottom of your change...
